### ISSUE TITLE
Can you explain the differences between your implementation and the official one?

### ISSUE THREAD

**Issue by gdwei** [Lornatang/UGATIT-PyTorch#4]
It would be appreciated if you can make it more clear in the readme.

**Comment by Lornatang** [comment]
Ha ha, I'm too lazy.

**Comment by gdwei** [comment]
A high level explanation is somewhat helpful for us to get started.

**Comment by Lornatang** [comment]
1- I use the light model architecture to reduce the number of model parameters;
2-I didn't use 256 * 256 images as input, I chose 96 * 96. Because I don't have a graphics card with more than 20GB of memory.
3-I didn't finish the full training, and the training iterations stopped training when it looked good.

**Comment by gdwei** [comment]
> 1- I use the light model architecture to reduce the number of model parameters;
> 2-I didn't use 256 * 256 images as input, I chose 96 * 96. Because I don't have a graphics card with more than 20GB of memory.
> 3-I didn't finish the full training, and the training iterations stopped training when it looked good.

Thanks, these descriptions really help.

To make it more clear, after reading your code,
I listed my understanding and questions below, pls point me out if I was wrong:

1. You reduce the number of model parameters by changing the light model:   nn.Linear(ngf * mult, ngf * mult, bias=False)   -->     nn.Linear(image_size * image_size * 16, 256, bias=False);

2. It seems that you did not do something special for 96*96 images in the code?

3. I do not understand what do you mean by 'the full training'. The training should be stopped when the results are visually looked satisfied, right?

**Comment by Lornatang** [comment]
Answer:
1-None
2-96 * 96 images have been completed in the process of processing my dataset. It's not part of the code.
3-According to the requirements of the original paper to training, about half a month. After several days of training, I print pictures every certain batch. After visualization, the training stopped subjectively.

**Comment by gdwei** [comment]
> Answer:
> 1-None
> 2-96 * 96 images have been completed in the process of processing my dataset. It's not part of the code.
> 3-According to the requirements of the original paper to training, about half a month. After several days of training, I print pictures every certain batch. After visualization, the training stopped subjectively.

Thanks for your patience.
Since it seems that you have get some reasonable results, so I have one last question:
I am training the network using my own dataset now,  the generator loss slowly goes down from around 1500 to about 50, but the discrimintor loss does not change a lot, it keep jittering aound.
![image](https://user-images.githubusercontent.com/8806352/95655233-670e5a00-0b38-11eb-8831-fff2e65b1253.png)
![image](https://user-images.githubusercontent.com/8806352/95655237-6d043b00-0b38-11eb-8c6b-f1440ea86f25.png)
Y axis: G_loss or D_loss
X axis: # of steps/iters

Is it normal? or how should these losses change according to time?

**Comment by Lornatang** [comment]
Keep training....

**Comment by gdwei** [comment]
> Keep training....

Ha ha, I know that it is not trained long enough, but I have actually trained it for around 10K iters, the discriminator loss kept looking like this. I just want to hear some experience from you guys who have successfully trained it, so that I can avoid waiting too much time on a network that will never give me the right results, because you know, we have to wait for a long time...
Anyway, if you have some suggestion on training or configuring this network for small image, pls tell me! 
I am trying to train on 3D image with size of 32x32x32.

**Comment by Lornatang** [comment]
It is well known that Gan is difficult to train~~~
